src/compiler/factory/nodeFactory.ts

```diff
@@ -1174,8 +1174,6 @@ export function createNodeFactory(flags: NodeFactoryFlags, baseFactory: BaseNode
                 if (elements.transformFlags === undefined) {
                     aggregateChildrenFlags(elements as MutableNodeArray<T>);
                 }
-                Debug.attachNodeArrayDebugInfo(elements);
-                return elements;

```

Comment:
This feels bit scary to me; I do think we need to think about how we should be defining what this function is actually supposed to do, i.e. come up with `create` and `update` separately or something (what you mention in #59137).